packages/css-framework/src/components/_switch.scss

```diff
@@ -54,68 +45,44 @@ $elements: 5;
     display: none;
   }
 
-  &::before {
-    content: attr(data-label);
-    display: inline-block;
-    width: auto;
-    height: auto;
-    padding: spacing(2) spacing(4);
-    font-size: inherit;
-    color: color(neutral, 400);
-    border: 1px solid $border-color;
-    border-right: none;
-    background-color: $background-color;
+  margin-right: -1px;
+  &:first-child {
+    border-radius: 3px 0 0 3px;
   }
-
-  &:hover {
-    &::before {
-      border-radius: $border-radius;
-      border-color: $background-color;
-      box-shadow: 0 0 0 1px $active-color;
-      background-color: $active-background-color;
-    }
+  &:last-child {
+    border-radius: 0 3px 3px 0;
+    margin-right: 0; 
   }
 
-  &.is-active {
-    z-index: $elements + 2;
-
-    &::before {
-      border: 1px solid $border-color;
-      border-radius: $border-radius;
-      border-color: $active-background-color;
-      box-shadow: 0 0 0 2px $active-color;
-      background-color: $active-background-color;
-    }
-  }
+  
 
-  &:hover, &.is-active {
-    &:first-of-type {
-      &::before {
-        border-bottom-right-radius: 0;
-        border-top-right-radius: 0;
-      }
-    }
-
-    &:last-of-type {
-      &::before {
-        border-top-left-radius: 0;
-        border-bottom-left-radius: 0;
-      }
-    }
+  display: inline-block;
+  width: auto;
+  height: auto;
+  padding: spacing(2) spacing(4);
+  font-size: inherit;
+  color: color(neutral, 400);
+  border: 1px solid $border-color;
+  background-color: $background-color;
+  position: relative;
+  z-index: 0;
```

Comment:
vars for z-index?

Reply:
Good spot.

For this I've made the `.rn-switch` component `position: relative`. This will automatically set the `z-index` to `auto` for this class. By doing this, it allows us to scope the stacking order of the `.rn-switch__option` to within the `.rn-switch`. If we'd used variables here, instead of `positon: relative` on the `.rn-switch`, then the `.rn-switch__option` would have displayed higher than the switch itself.